storage/types.go

```diff
@@ -0,0 +1,52 @@
+package storage
+
+import (
+	"io"
+
+	"github.com/gnolang/gno/tm2/pkg/bft/types"
+)
+
+// Storage represents the permanent storage abstraction
+// for reading and writing operations
+type Storage interface {
+	io.Closer
+	Reader
+	Writer
+}
+
+// StorageRead defines the transaction storage interface for read methods
+type Reader interface {
+	io.Closer
+	// GetLatestHeight returns the latest block height from the storage
+	GetLatestHeight() (int64, error)
+
+	// GetBlock fetches the block by its number
+	GetBlock(int64) (*types.Block, error)
+
+	// GetTx fetches the tx using its hash
+	GetTx([]byte) (*types.TxResult, error)
+}
+
+// StorageWrite defines the transaction storage interface for write methods
```

Comment:
Nitpick: the comment doesn't match the interface name, probably leftover

Reply:
Resolved in [1a82f41](https://github.com/gnolang/tx-indexer/pull/8/commits/1a82f4175f3bbc267786ab215e4b69760c998958)